build-scripts/runtime_lib.cmake

```diff
@@ -1,6 +1,46 @@
 # Copyright (C) 2019 Intel Corporation. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+file(WRITE "${PROJECT_BINARY_DIR}/try_compile_clock_nanosleep.c"
```

Comment:
Compiling the files here may be a little heavy, it will be performed even when libc-wasi isn't enabled. How about moving the changes to `core/iwasm/libraries/libc-wasi/libc_wasi.cmake`?

Reply:
Sure, moved it